pkg/kubelet/server/auth.go

```diff
@@ -81,35 +83,52 @@ func (n nodeAuthorizerAttributesGetter) GetRequestAttributes(u user.Info, r *htt
 
 	requestPath := r.URL.Path
 
-	// Default attributes mirror the API attributes that would allow this access to the kubelet API
-	attrs := authorizer.AttributesRecord{
-		User:            u,
-		Verb:            apiVerb,
-		Namespace:       "",
-		APIGroup:        "",
-		APIVersion:      "v1",
-		Resource:        "nodes",
-		Subresource:     "proxy",
-		Name:            string(n.nodeName),
-		ResourceRequest: true,
-		Path:            requestPath,
+	var subresources []string
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletFineGrainedAuthz) {
+		switch {
+		case isSubpath(requestPath, podsPath):
+			subresources = append(subresources, "pods")
+		case isSubpath(requestPath, runningPodsPath):
```

Comment:
nit, put runningpods last since it will alloc a new string on every check since the handler path has a trailing slash

Reply:
Done.